rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -96,13 +96,11 @@ public T doExecute(Object... args) throws Throwable {
                         break;
                     }
 
-                    if (RootContext.inGlobalTransaction()) {
-                        //do as usual
+                    if (RootContext.inGlobalTransaction() || RootContext.requireGlobalLock()) {
+                        // do as usual
+                        // in @GlobalLock env, don't need to append key eithor,
+                        // but check lock hear, do the same thing as @GlobalTransactional env
```

Comment:
Is it better to comment the code like **`"Do the same thing under either @GlobalTransactional or @GlobalLock, that only check the global lock here."`** or something else more accurate? There is nothing about **`“key appending”`** in the new code block, so the comment could be a little puzzling.